lib/interactor/context.rb

```diff
@@ -126,6 +126,33 @@ def fail!(context = {})
       raise Failure, self
     end
 
+    # Public: Halt the Interactor::Context. Halting a context allows stopping the
+    # context without failing it. It is often used in stopping Organizer chains
+    # without explicitly failing. The context is not flagged as having failed.
+    #
+    # Optionally the caller may provide a hash of key/value pairs to be merged
+    # into the context before halting.
+    #
+    # context - A Hash whose key/value pairs are merged into the existing
+    #           Interactor::Context instance. (default: {})
+    #
+    # Examples
+    #
+    #   context = Interactor::Context.new
+    #   # => #<Interactor::Context>
+    #   context.halt!
+    #   # => Interactor::Halt: #<Interactor::Context>
+    #   context.halt! rescue false
+    #   # => false
+    #   context.halt!(foo: "baz")
+    #   # => Interactor::Halt: #<Interactor::Context foo="baz">
+    #
+    # Raises Interactor::Halt initialized with the Interactor::Context.
+    def halt!(context = {})
+      modifiable.update(context)
+      raise Halt, self
```

Comment:
@gregory Mainly to stick within convention of the rest of the gem. I agree it's probably a more accurate usage here, but wanted to keep consistency since benefit was not major.